lib/at_ex/gateway/Voice/upload_media.ex

```diff
@@ -1,2 +1,45 @@
 defmodule AtEx.Gateway.Voice.UploadMedia do
+  import AtEx.Util
+
+  @live_url "https://voice.africastalking.com"
+  @sandbox_url "https://voice.sandbox.africastalking.com"
+
+  use AtEx.Gateway.Base, url: get_url(@live_url, @sandbox_url)
+
+  @doc """
+  This function makes a POST request to upload a media in the Africa's talking queue status endpoint, this
+  function accepts a map of parameters.
+  sent
+
+  ## Parameters
+  attrs: - a map containing:
+    - `username` - Your Africa’s Talking application username. Can be set in the config
+    - `url` - The url of the file to upload. Don’t forget to start with http://
+
+
+  ## Example
+      iex> AtEx.Gateway.Voice.UploadMedia.upload(%{
+               phoneNumber: "+254728833180",
+        ...>   url: "https://www.soundhelix.com/examples/mp3/SoundHelix-Song-8.mp3"
+        ...> })
+        {:ok, result}
+  """
+
+  def upload(attrs) do
+    username = Application.get_env(:at_ex, :username)
+
+    params =
+      attrs
+      |> Map.put(:username, username)
+
+    with {:ok, %{status: 201} = res} <- post("/mediaUpload", params) do
```

Comment:
@kamalogudah i think you should use `process_result()` here.
Everything else looks fine

Reply:
Was to use it but this returns, a string as the body, which was returning a Jason parse error. Hence decided to handle the reponse this way.